libp2p/network/swarm.py

```diff
@@ -229,17 +228,8 @@ def set_stream_handler(self, stream_handler: StreamHandlerFn) -> None:
     async def close(self) -> None:
         # TODO: Prevent from new listeners and conns being added.
```

Comment:
I mentioned this in my other review but I think the way this can be accomplished is to implement an equivalent `external_api` decorator for the trio-based service and wrap all of the external methods of this class in it so that when called after closing they will fail.